pkg/apiserver/handlers/webhook/mutation_crd.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	admv1 "k8s.io/api/admission/v1"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/controller/networkpolicy"
+)
+
+func HandleMutationNetworkPolicy(m *networkpolicy.NetworkPolicyMutator) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		klog.V(2).Info("Received request to mutate Antrea Policy CRD")
+		var reqBody []byte
+		if r.Body != nil {
+			reqBody, _ = ioutil.ReadAll(r.Body)
+		}
+		if len(reqBody) == 0 {
+			klog.Errorf("Mutation webhook crdmutator received empty request body")
+			http.Error(w, "empty request body", http.StatusBadRequest)
+			return
+		}
+		// verify the content type is accurate
+		contentType := r.Header.Get("Content-Type")
```

Comment:
I'm curious. Aren't there utilities we can use in `github.com/kubernetes-sigs/controller-runtime` to write validation & mutation webhooks, so we don't have to deal with the raw HTTP request? More of a question for @tnqn and @abhiraut. I am not saying this needs to be addressed in this PR.

Reply:
kubebuilder is one way.. i did not easily find a way to inject webhook functions in controller-runtime.

one possible refactor is to use a single HandleWebhooks function to handle both, mutation and validation requests